### PR TITLE
Filters alerts integration

### PR DESCRIPTION
**Pull request by MiguelAngelGutierrezMaya**
## Description

This PR connects expense filtering in the dashboard with backend query support, and adds the monitoring and alerting foundation for the platform. It also introduces retry behavior for transient client API failures and X-Ray tracing across the main services so the new alarms have enough signal to point to the failing component.

### Core changes

- Added optional expense filters (`expense_type_id`, `expense_category_id`, `name`) from shared models through the expenses service and repository, including filtered counts and cursor pagination.
- Added the dashboard filtering experience with `SearchInput`, `FilterChip`, `FilterBar`, filter loading state, and filter-aware pagination so the expenses list refreshes without losing catalog context.
- Added retry with exponential backoff and jitter in the dashboard API client for transient network, timeout, and `429`/`5xx` responses.
- Added a new `@packages/notifications` Lambda that consumes CloudWatch alarm notifications from SNS and sends formatted SES emails using the new `service-alert` transactional templates in English and Spanish.
- Added the v3 monitoring stack with a CloudWatch dashboard, alarms for API Gateway, Lambda services, Cognito triggers, and the notification Lambda wiring.
- Added Powertools tracing across API Gateway/Lambda entrypoints, repositories, and database access to improve request-level observability.

### Schema changes

- No database schema or migration changes were introduced in this PR.
- `GET /expenses` now accepts the optional query params `expense_type_id`, `expense_category_id`, and `name`.
- Infra/runtime configuration now expects `EMAILS_PREFIX`, `ALERT_EMAIL_TO`, and `ALERT_EMAIL_FROM`.

### Minor changes

- Renamed `COGNITO_EMAILS_PREFIX` to `EMAILS_PREFIX` across defaults, upload scripts, and deployment workflows.
- Exported cross-version identifiers from existing stacks so the new monitoring stack can reference API Gateway, Amplify, Lambda services, and Cognito triggers.
- Added and updated coverage across models, dashboard, UI, expenses service, notifications, transactional emails, and infra tests.

## Screenshots/Recordings

N/A

## How to Test

1. Configure the local env with `EMAILS_PREFIX`, `ALERT_EMAIL_TO`, and `ALERT_EMAIL_FROM`.
2. Start the app with `pnpm --filter @client/main web` and run the local API/services setup you normally use for the expenses flow.
Expected result: the dashboard loads successfully and the expenses page can fetch data.
3. Open the Expenses page and confirm the filter bar is rendered after catalogs load, with a search field, expense type chips, and expense category chips.
Expected result: the filters are visible before interacting with the list.
4. Apply each filter independently.
Expected result: the list only shows matching expenses, the loading state is transient, and the total/count remains consistent with the filtered dataset.
5. Combine filters by type, category, and partial name search, then paginate the list.
Expected result: filters are preserved while loading more results, and the next page only contains records that still match the active filters.
6. Clear the search text and toggle the selected chips off.
Expected result: the full expenses list is restored without stale or duplicated items.
7. Run `pnpm email:build && pnpm email:export`.
Expected result: the `service-alert` email is generated for both `en` and `es`.
8. Run `pnpm infra:cdk synth`.
Expected result: synthesis succeeds and includes the v3 monitoring stack, the notification Lambda, the SNS topic, and the monitoring resources wired to the exported v1/v2 identifiers.

## Checklist

- [x] The PR description includes clear instructions on how to test the implementation manually.

##### Added Tests?

- [x] yes
  - [x] Verified that this PR is not creating duplicate use cases
  - [x] All edge cases described in the acceptance criteria are covered, and tests have clear and descriptive names
- [ ] no, because they aren't needed
- [ ] no, because I need help

_Override these when needed._

```sh

```

### CODE REVIEW

File: client/packages/features/dashboard/src/presentation/providers/expense-provider.tsx [hunk 1]
```diff
@@ -109,42 +117,60 @@ export function ExpenseProvider({
...
+        );
+      } finally {
+        if (!controller.signal.aborted && mountedRef.current) {
+          setInitialLoading(false);
+          setFiltering(false);
+        }
+      }
+    },
+    [repository],
+  );
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply client/packages/features/dashboard/src/presentation/providers/expense-provider.tsx:161]: This change is not necessary; adding this could cause unnecessary re-renders

File: client/packages/features/dashboard/src/presentation/providers/expense-provider.tsx [hunk 2]
```diff
@@ -159,6 +185,15 @@ export function ExpenseProvider({
     }
   }, [nextCursor, loadingMore, repository]);
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply client/packages/features/dashboard/src/presentation/providers/expense-provider.tsx:186]: This change is not necessary; adding this could cause unnecessary re-renders
